### DISCUSSION TITLE
Help in python code

### DISCUSSION THREAD

**Discussion by natti-cyber** [community/community#21492] [Other Feature Feedback, Questions, & Ideas]
<p>Hy I am new in python and I stuck in this code. I didn’t get how the second part after while loop worked. would anyone explain to me please.<br>
Thank you!</p>
<p>n = 6<br>
a , b, c = 1, 1, 1<br>
cnt = 2<br>
while (cnt &lt; n):<br>
cnt += 1<br>
a = b + c + 1<br>
c = b<br>
b = a<br>
print(a)</p>
<p>the answer it return is 15 and I didn’t get it how.</p>

**Comment by JPConnellan** [comment, marked as answer]
<p>If you carefully watch each variable as you go through the loop, you will see they change like this:</p>
<p>cnt = 3<br>
a=3<br>
b=3<br>
c=1</p>
<p>cnt = 4<br>
a=5<br>
b=5<br>
c=3</p>
<p>cnt = 5<br>
a=9<br>
b=9<br>
c=5</p>
<p>cnt = 6<br>
a=15<br>
b=15<br>
c=9</p>
<p>The reason the last iteration runs is because before the while loop was entered, cnt was 5, thereby satisfying the condition cnt &lt; 6</p>

**Comment by natti-cyber** [comment]
<p>I totally got it. Thank you so much :slight_smile:</p>
